Applications/SlicerApp/Testing/Python/CLISerializationTest.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python
+
+#
+#  Program: 3D Slicer
+#
+#  Copyright (c) Kitware Inc.
+#
+#  See COPYRIGHT.txt
+#  or http://www.slicer.org/copyright/copyright.txt for details.
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This file was originally developed by Johan Andruejol, Kitware Inc.
+#
+
+import argparse
+import json
+import os
+import sys
+
+from SlicerAppTesting import *
+
+"""
+Usage:
+    CLISerializationTest.py
+      /path/to/Slicer /path/to/CLIExecutables /path/to/data_dir /path/to/temp_dir
+"""
+class CLISerializationTest(object):
+  def __init__(self):
+    self.SlicerExecutable = None
+    self.CLIDir = None
+    self.DataDir = None
+    self.TempDir = None
+    self.CLIName = None
+    self.CLIRequiredInputs = []
+    self.CLIOptionnalInputs = []
+    self.ExpectedJSON = None
+    self.Success = False
+
+  def __del__(self):
+    if self.Success:
+      try:
+        os.remove(self.JSONFilePath)
+      except AttributeError, OSError:
+        pass
+
+  def runTest(self):
+    # Run the test of serializing/deserializing a CLI
+    slicer_base_dir = os.path.dirname(self.SlicerExecutable)
+
+    # Find the cli executable
+    cli_executable = '%s/%s/%s.exe' %(slicer_base_dir, self.CLIDir, self.CLIName)
```

Comment:
Since all CLIs are in the path, you should be able to directly execute `NameOfCLI` or `NameOfCLI.exe`